src/loaders/common/loader_helper.py

```diff
@@ -179,17 +181,37 @@ def is_upa_info_complete(upa_dir: str):
     return True
 
 
+def make_job_dir(root_dir, job_dir, username):
+    """Helper function that creates a job_dir for a user under root directory."""
+    job_dir = os.path.join(root_dir, job_dir, username)
+    os.makedirs(job_dir, exist_ok=True)
+    # only user can cread, write, or execute
+    os.chmod(job_dir, stat.S_IRWXU)
+    return job_dir
+
+
+def make_output_dir(root_dir, source_data_dir, source, env, workspace_id):
+    """Helper function that makes output directory for a specific collection under root directory."""
+    output_dir = os.path.join(root_dir, source_data_dir, source, env, str(workspace_id))
+    os.makedirs(output_dir, exist_ok=True)
+    return output_dir
+
+
 def make_collection_source_dir(
         root_dir: str,
         env: str,
         collection: str,
-        source_ver: str
+        source_ver: str,
+        upload: bool = False,
 ) -> str:
     """
     Helper function that creates a collection & source_version and link in data
     to that collection from the overall source data dir.
     """
-    csd = os.path.join(root_dir, loader_common_names.COLLECTION_SOURCE_DIR, env, collection, source_ver)
+    if upload:
+        csd = os.path.join(root_dir, loader_common_names.COLLECTION_SOURCE_DIR, env, "upload", collection, source_ver)
```

Comment:
Please correct me if I am wrong @MrCreosote.

So you are thinking we can just use a different `collection` name instead? Like `GTDB_UPLOAD` rather than adding an extra `upload` directory?

Reply:
I think you can use the same collection name, since they'll have different env values (see another comment below)

Reply:
Updated!

Reply:
:+1: